telegram/ext/_updater.py

```diff
@@ -17,174 +17,146 @@
-        'running',
-        'is_idle',
-        'httpd',
+        '_last_update_id',
+        '_running',
+        '_initialized',
+        '_httpd',
         '__lock',
-        '__threads',
+        '__polling_task',
     )
 
     def __init__(
-        self: 'Updater[BT, DT]',
-        *,
-        user_signal_handler: Callable[[int, object], Any] = None,
-        dispatcher: DT = None,
-        bot: BT = None,
-        update_queue: Queue = None,
-        exception_event: Event = None,
+        self,
+        bot: 'Bot',
+        update_queue: asyncio.Queue,
     ):
-        if not was_called_by(
-            inspect.currentframe(), Path(__file__).parent.resolve() / '_builders.py'
-        ):
-            warn(
-                '`Updater` instances should be built via the `UpdaterBuilder`.',
-                stacklevel=2,
-            )
-
-        self.user_signal_handler = user_signal_handler
-        self.dispatcher = dispatcher
-        if self.dispatcher:
-            self.bot = self.dispatcher.bot
-            self.update_queue = self.dispatcher.update_queue
-            self.exception_event = self.dispatcher.exception_event
-        else:
-            self.bot = bot
-            self.update_queue = update_queue
-            self.exception_event = exception_event
-
-        self.last_update_id = 0
-        self.running = False
-        self.is_idle = False
-        self.httpd = None
-        self.__lock = Lock()
-        self.__threads: List[Thread] = []
-        self.logger = logging.getLogger(__name__)
+        self.bot = bot
+        self.update_queue = update_queue
+
+        self._last_update_id = 0
+        self._running = False
+        self._initialized = False
+        self._httpd: Optional[WebhookServer] = None
+        self.__lock = asyncio.Lock()
+        self.__polling_task: Optional[asyncio.Task] = None
+        self._logger = logging.getLogger(__name__)
+
+    @property
+    def running(self) -> bool:
+        return self._running
+
+    async def initialize(self) -> None:
+        if self._initialized:
+            self._logger.debug('This Updater is already initialized.')
+            return
+
+        await self.bot.initialize()
+        self._initialized = True
+
+    async def shutdown(self) -> None:
+        """
 
-    @staticmethod
-    def builder() -> 'InitUpdaterBuilder':
-        """Convenience method. Returns a new :class:`telegram.ext.UpdaterBuilder`.
+        Returns:
 
-        .. versionadded:: 14.0
+        Raises:
+            :exc:`RuntimeError`: If the updater is still running.
         """
-        # Unfortunately this needs to be here due to cyclical imports
-        from telegram.ext import UpdaterBuilder  # pylint: disable=import-outside-toplevel
+        if self.running:
+            raise RuntimeError('This Updater is still running!')
 
-        return UpdaterBuilder()
+        if not self._initialized:
+            self._logger.warning('This Updater is already shut down.')
```

Comment:
change from warning to debug level. Same for Bot, App, and HTTPXRequest, and wherever there's a shutdown()